posthog/temporal/data_imports/pipelines/pipeline/delta_table_helper.py

```diff
@@ -106,7 +110,7 @@ def write_to_deltalake(
         if delta_table:
             delta_table = self._evolve_delta_schema(data.schema)
 
-        if is_incremental and delta_table is not None:
+        if is_incremental and delta_table is not None and not self._is_first_sync:
```

Comment:
Given that `delta_table` is coming from `get_delta_table`, where `_is_first_sync` is only set to `True` when `None` is returned, it would appear that `_is_first_sync` will always be `True` when `delta_table` is `None` and always be `False` when `delta_table` is not `None`, thus this check seems pointless: If 
`delta_table` is not `None`, how can `_is_first_sync` be anything but `False`?

Unless I'm missing something?

Reply:
`_is_first_sync` can also be set from the `reset()` function too. On a second chunk, `delta_table` will never be `None` and thus we still want to check `_is_first_sync`. The `write` method is called multiple times over the life of a sync

Reply:
Ah, that makes sense, thanks for the additional context!